public/js/views/management/confirm-del-pay-method.jsx

```diff
@@ -0,0 +1,204 @@
+import React, { Component, PropTypes } from 'react';
+
+import $ from 'jquery';
+import PayMethodItem from 'components/pay-method-item';
+import PayMethodDropDown from 'components/pay-method-drop-down';
+import SubmitButton from 'components/submit-button';
+import Subscription from 'components/subscription';
+import Spinner from 'components/spinner';
+
+import { default as tracking } from 'tracking';
+import { gettext, getPayMethodFromUri, setTitle } from 'utils';
+
+
+export default class ConfirmDelPayMethod extends Component {
+
+  static propTypes = {
+    affectedSubscriptions: PropTypes.array,
+    closeModal: PropTypes.func.isRequired,
+    delPayMethod: PropTypes.func.isRequired,
+    error: PropTypes.func.isRequired,
+    getFilteredUserSubscriptions: PropTypes.func.isRequired,
+    payMethodUri: PropTypes.string.isRequired,
+    payMethods: PropTypes.array.isRequired,
+    showDelPayMethod: PropTypes.func.isRequired,
+    showPayMethods: PropTypes.func.isRequired,
+    updateSubPayMethod: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      payMethod: getPayMethodFromUri(
+        this.props.payMethods, this.props.payMethodUri),
+    };
+  }
+
+  componentDidMount() {
+    setTitle('Delete Payment Method?');
+    tracking.setPage('/confirm-del-pay-method');
+    this.props.getFilteredUserSubscriptions(this.props.payMethodUri);
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.setState({isSubmitting: true});
+    console.log(e);
+
+
+    var affectedSubs = this.props.affectedSubscriptions;
+    var deferreds = [];
+    // If there are affected subs...
+    if (affectedSubs) {
+      var select = e.target.querySelector('[name="new-pay-method"]');
+      //... and there's a select for choosing the new pay method
+      console.log(select);
+      if (select && select.options) {
+        var newPayMethodUri = select.options[select.selectedIndex].value;
+        console.log(newPayMethodUri);
+        // Update the list of subs to use newPayMethodUri
+        for (var i = 0; i < affectedSubs.length; i += 1) {
+          var sub = affectedSubs[i];
+          deferreds.push(
+            this.props.updateSubPayMethod(sub.resource_uri, newPayMethodUri));
+        }
+      }
+    }
+
+    // Delete the card in question.
+    deferreds.push(this.props.delPayMethod(this.props.payMethodUri));
+
+    $.when(...deferreds).done(() => {
+      this.props.showPayMethods();
+    }).fail(() => {
+      this.props.error('Failed to update subscriptions');
+    });
```

Comment:
I think this will all be easier to test if it's encapsulated within the action. I know it will probably be simpler when the API changes though.